File: Napakalaki_PDOO_P5_S1-2-3_JAVA/src/NapakalakiGame/Cultist.java
package NapakalakiGame;

public class Cultist implements Card{
	private String name;
	private int gainedLevels;
	
	/**
	 * Constructor por defecto
	 * @param name
	 * @param gainedLevels
	 */
	public Cultist(String pName, int pGainedLevels) {
		this.name = pName;
		this.gainedLevels = pGainedLevels;
	}

	/**
	 * @return el parametro nombre
	 */
	public String getName() {
		return name;
	}

	/**
	 * @return el parametro Niveles Ganados
	 */
	public int getGainedLevels() {
		return this.gainedLevels;
	}

	@Override
	public int getBasicValue() {
		return this.gainedLevels;
	}

	@Override
	public int getSpecialValue() {
		return this.getBasicValue()*CultistPlayer.getTotalCultistPlayers();
	}
	
}

File: 02Segundo/Programacion_Diseno_Orientada_Objetos_PDOO/Rupakalaki/lib/napakalaki/cultist.rb
module NapakalakiGame

  class Cultist
    attr_accessor :name, :gainedLevels
    
    def initialize(name,gainedLevels)
      @name = name
      @gainedLevels = gainedLevels
    end

    def basicValue()
        return @gainedLevels;
    end

    def specialValue()
      return basicValue() * CultistPlayer.totalCultistPlayers();
    end

    def to_s
      return " Nombre: #{@name} Tipo: #{@gainedLevels}"
    end

    public_class_method :new
  end
  
end
